runsc/fsgofer/fsgofer.go

```diff
@@ -1026,7 +1059,7 @@ func (l *localFile) Flush() error {
 
 // Connect implements p9.File.
 func (l *localFile) Connect(p9.ConnectFlags) (*fd.FD, error) {
-	return nil, syscall.ECONNREFUSED
+	return fd.DialUnix(l.hostPath)
```

Comment:
Let's check for HostUDSAllowed flag here and return ECONNREFUSED if not allowed. Otherwise it may allow UDS on existing files inside mounts.